### DISCUSSION TITLE
Migration to Java 17 using OpenRewrite UpgradeToJava17

### DISCUSSION THREAD

**Discussion by zachtyll** [runelite/runelite#17797] [Ideas]
Hey guys, I tried using [OpenRewrite](https://docs.openrewrite.org/running-recipes/popular-recipe-guides/migrate-to-java-17) for Maven to upgrade Runelite to Java 17.

My goal was to log on to OSRS using Runelite running on the Java 17 runtime and in the end i got a messy PoC working for that goal. My changes are visible in this repo: [https://github.com/zachtyll/runelite](https://github.com/zachtyll/runelite).

I had to remove some code and modify some things relating to file reading to make it start. I don't know if that will cause major regressions somewhere, but it is at least possible to log on and walk around and so on.

It would be cool to hear what could go wrong after the changes, if it breaks API used by plugins and so on.

**Comment by Adam-** [comment]
Hello. We ship our own Java runtime, so that users do not have to separately install Java, and the version we ship on Windows currently is 11. So we can't update the language level the client uses until we know that every launcher is running Java 17. We might do this at some point but currently I have other higher priorities.
